### ISSUE TITLE
Add support for Twig inside Markdown (Twig-Markdown)

### ISSUE THREAD

**Issue by mbirth** [tpmatthes/Twig.novaextension#8]
Could you maybe add support for Twig inside Markdown (e.g. the GRAV CMS uses this)? Thanks a lot!

**Comment by tpmatthes** [comment]
Hey Markus, thanks for your feature request! I’m happy to look into it. However, since I’m not familiar with Grav CMS, can you give me a code example for this? 😊

**Comment by mbirth** [comment]
Sure, a page could look like this:

```markdown
---
title: Example page
twig_first: true
process:
  markdown: true
  twig: true
modified_date: '2019-01-26 14:54:40'
---

# Welcome to my example

This is some text in *Markdown*. Yet Twig and HTML is also possible as you can see now:

{% set allp = allallpages() %}
{{ allp|count() }} pages found.

{% for child in allp %}
  <li><a href="{{ child.url }}">{{ child.title }} ({{child.language}})</a></li>
{% endfor %}
```

**Comment by tpmatthes** [comment]
Thank you! I could probably create a subsyntax for Markdown.

Which suffixes does Grav CMS support for markdown files?

**Comment by mbirth** [comment]
Grav uses `.md`, so normal Markdown suffix.

**Comment by tpmatthes** [comment]
Ok, this may be a problem, because multiple template engines use `{%` and I can’t just overwrite the default syntax highlighting for Markdown for all users. Is it possible to use other, more specific suffixes like `.twig.md`?

If this isn’t possible, I could add a “hidden” syntax definition to my plugin, which would allow you to map “.md” to this new syntax definition in Nova’s settings. What do you think?

**Comment by mbirth** [comment]
Yes, I think the hidden definition is the best solution to not cause confusion when opening other Markdown files.

**Comment by tpmatthes** [comment]
Alright, I’ve uploaded a pre-release for version 1.7 of my plugin with a new syntax definition called “Markdown-Twig”. It extends the default Markdown definition and adds syntax highlighting for Twig, so you can use it for all Markdown files. Can you test it?

How to test it:

1. Deactivate the “Twig” plugin in Nova’s plugin manager.
2. Download the source code (zip) for the pre-release: https://github.com/tpmatthes/Twig.novaextension/releases/tag/1.7
3. Unpack the zip archive.
4. Rename the unpacked folder to “Twig.novaextension”.
5. Open the folder with Nova and install the plugin.
6. Map “.md” to “Twig-Markdown” in the settings menu.
![settings](https://user-images.githubusercontent.com/30212990/126314593-bd28d7e1-9843-467d-8e2f-32a7dafcd8b7.png)

Don’t forget to remove the pre-release from Nova and reinstall the original plugin afterwards!

**Comment by mbirth** [comment]
Sorry, didn't find time to test it earlier. But it looks like it works great:

<img width="665" alt="image" src="https://user-images.githubusercontent.com/130302/127752359-520d47cc-a610-4b33-b037-6dd502669a55.png">

**Comment by tpmatthes** [comment]
No problem, thanks for testing the pre-release version! A quick question about your screenshot: Why are the Twig brackets colored differently? Is this influenced by a setting/plugin or a bug with the syntax highlighting? 🤔

**Comment by mbirth** [comment]
> Why are the Twig brackets colored differently? Is this influenced by a setting/plugin or a bug with the syntax highlighting?

I'm pretty sure this is from the "Use rainbow brackets" option in Nova's preferences. 😉

**Comment by tpmatthes** [comment]
Ah ok! This setting doesn’t apply in .twig files, but in .md, which is a little bit confusing. I’ve fixed this and released version 1.7 of my plugin. You can install it via Nova’s plugin manager. 😊
